### PR TITLE
Added player positions synchronisation over network.

### PR DESCRIPTION
**Pull request by Amyantis**

### CODE REVIEW

File: client/src/MovementDispatcher.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "NetworkThread.h"
+
+class MovementDispatcher {
+ private:
+  const int32_t refreshRateMs;
+  sf::Clock clock;
+  bool needUpdate;
+
+ public:
+  explicit MovementDispatcher(const int32_t refreshRateMs = 50);
```
Comment (Shiroy) [inline client/src/MovementDispatcher.h]: You can set the refresh rate at 1s

It's not necessary to overload th network now.

File: server/src/Server.cpp [hunk 2]
```diff
@@ -32,8 +32,22 @@ void Server::run() {
       m_listenerThread.getSessionToAdd().pop();
 
       auto newPlayer = EntityManager::getInstance().createPlayer("Player", newSession);
-      newPlayer.lock()->addToWorld();
+      const std::shared_ptr<Player> &newPlayer_shrd = newPlayer.lock();
+      newPlayer_shrd->addToWorld();
       newSession->setPlayer(newPlayer);
+
+      for (auto entity: EntityManager::getInstance().getAllEntities()) {
```
Comment (Shiroy) [inline server/src/Server.cpp]: Move that to a function :)

File: client/src/MovementDispatcher.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+#include <iostream>
+#include "MovementDispatcher.h"
+
+MovementDispatcher::MovementDispatcher(const int32_t refreshRateMs)
+    : refreshRateMs(refreshRateMs),
+      needUpdate(false) {}
+
+void MovementDispatcher::updateNetwork(const PlayerCommands &playerCommands,
+                                       LockedQueue<AresProtocol::AresMessage> &sendingQueue) {
+  sf::Time deltaTime = clock.restart();
```
Comment (Shiroy) [inline client/src/MovementDispatcher.cpp]: If the loop is always faster thant the refreshRate, the movements are never broadcast to the network.

Use [sf::Clock::getElapsedTime](https://www.sfml-dev.org/documentation/2.0-fr/classsf_1_1Clock.php#a799feb6acb099b57b58d8d20984fce11) instead.

File: client/src/MovementDispatcher.h (hunk 1 above)
Comment (Shiroy) [reply client/src/MovementDispatcher.h]: Also, move the `50` constant to a `#define`

**Review by Shiroy — CHANGES_REQUESTED** [review]

File: client/src/MovementDispatcher.cpp (hunk 3 above)
Comment (Amyantis) [reply client/src/MovementDispatcher.cpp]: Good catch!

**Review by Shiroy — APPROVED** [review]

**Pull request merged**